### PR TITLE
rocks: '--server' flag now accepts several arguments

### LINKED ISSUE #686: tt rocks does not respect --server option (version 2.0.0)
# Reproducer

Consider running this install:

```
❯ tt version
Tarantool CLI version 2.0.0, darwin/arm64. commit: d894732

❯ tt rocks version
tt rocks 3.9.2
LuaRocks package manager

❯ tt rocks --tree .rocks --lua-version 5.1 --server https://moonlibs.github.io/rocks install 'https://raw.githubusercontent.com/moonlibs/rocks/master/bin-scm-4.rockspec'
Cloning into 'bin'...
remote: Enumerating objects: 179, done.
remote: Counting objects: 100% (78/78), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 179 (delta 45), reused 55 (delta 35), pack-reused 101
Receiving objects: 100% (179/179), 63.11 KiB | 936.00 KiB/s, done.
Resolving deltas: 100% (92/92), done.
Missing dependencies for bin scm-4:
   ffi-reloadable >= 0 (not installed)

bin scm-4 depends on lua ~> 5.1 (5.1-1 provided by VM)
bin scm-4 depends on ffi-reloadable >= 0 (not installed)

Error: Could not satisfy dependency ffi-reloadable >= 0: No results matching query were found for Lua 5.1.
```

At the same time vanila Luarocks 3.9.2 respects flag --server as "append to rock_servers":

```
❯ luarocks --tree .rocks --lua-version 5.1 --server https://moonlibs.github.io/rocks install 'https://raw.githubusercontent.com/moonlibs/rocks/master/bin-scm-4.rockspec'
Cloning into 'bin'...
remote: Enumerating objects: 179, done.
remote: Counting objects: 100% (78/78), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 179 (delta 45), reused 55 (delta 35), pack-reused 101
Receiving objects: 100% (179/179), 63.11 KiB | 923.00 KiB/s, done.
Resolving deltas: 100% (92/92), done.
Missing dependencies for bin scm-4:
   ffi-reloadable >= 0 (not installed)

bin scm-4 depends on lua ~> 5.1 (5.1-1 provided by VM)
bin scm-4 depends on ffi-reloadable >= 0 (not installed)
Installing https://moonlibs.github.io/rocks/ffi-reloadable-scm-1.rockspec
Cloning into 'ffi-reloadable'...
remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 10 (delta 0), reused 1 (delta 0), pack-reused 7
Receiving objects: 100% (10/10), done.

ffi-reloadable scm-1 depends on lua >= 5.1 (5.1-1 provided by VM)
No existing manifest. Attempting to rebuild...
ffi-reloadable scm-1 is now installed in /Users/ochaton/personal/moonlibs/rocks/.rocks (license: Artistic)

env MACOSX_DEPLOYMENT_TARGET=11.0 gcc -O2 -fPIC -I/opt/homebrew/include/luajit-2.1 -c libluabin.c -o libluabin.o
env MACOSX_DEPLOYMENT_TARGET=11.0 gcc  -bundle -undefined dynamic_lookup -all_load -o libluabin-scm-4.so libluabin.o
bin scm-4 is now installed in /Users/ochaton/personal/moonlibs/rocks/.rocks (license: Artistic)
```

But it is still possible to create directory `.rocks` and file `config-5.1.lua` with following content:

```
❯ cat .rocks/config-5.1.lua
rocks_servers = {
    "http://moonlibs.github.io/rocks", -- moonlibs libs
}
```

And rocks_server will be respected and library will be successfully installed

```
❯ tt rocks --tree .rocks install 'https://raw.githubusercontent.com/moonlibs/rocks/master/bin-scm-4.rockspec'
Cloning into 'bin'...
remote: Enumerating objects: 179, done.
remote: Counting objects: 100% (78/78), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 179 (delta 45), reused 55 (delta 35), pack-reused 101
Receiving objects: 100% (179/179), 63.53 KiB | 834.00 KiB/s, done.
Resolving deltas: 100% (92/92), done.
Missing dependencies for bin scm-4:
   ffi-reloadable >= 0 (not installed)

bin scm-4 depends on lua ~> 5.1 (5.1-1 provided by VM)
bin scm-4 depends on ffi-reloadable >= 0 (not installed)
Installing http://moonlibs.github.io/rocks/ffi-reloadable-scm-1.rockspec
Cloning into 'ffi-reloadable'...
remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 10 (delta 0), reused 1 (delta 0), pack-reused 7
Receiving objects: 100% (10/10), done.

ffi-reloadable scm-1 depends on lua >= 5.1 (5.1-1 provided by VM)
No existing manifest. Attempting to rebuild...
ffi-reloadable scm-1 is now installed in /Users/ochaton/personal/moonlibs/rocks/.rocks (license: Artistic)

env MACOSX_DEPLOYMENT_TARGET=11.0 gcc -O2 -fPIC -I/Users/ochaton/tt-apps/include/include/tarantool -c libluabin.c -o libluabin.o
env MACOSX_DEPLOYMENT_TARGET=11.0 gcc  -bundle -undefined dynamic_lookup -all_load -o libluabin-scm-4.so libluabin.o
bin scm-4 is now installed in /Users/ochaton/personal/moonlibs/rocks/.rocks (license: Artistic)
```

Moreover `tt rocks` seems to be using flag `--only-server` in proper way:
```
❯ tt rocks --tree .rocks --lua-version 5.1 --only-server https://moonlibs.github.io/rocks install 'https://raw.githubusercontent.com/moonlibs/rocks/master/bin-scm-4.rockspec'
Cloning into 'bin'...
remote: Enumerating objects: 179, done.
remote: Counting objects: 100% (78/78), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 179 (delta 45), reused 55 (delta 35), pack-reused 101
Receiving objects: 100% (179/179), 63.53 KiB | 916.00 KiB/s, done.
Resolving deltas: 100% (92/92), done.
Missing dependencies for bin scm-4:
   ffi-reloadable >= 0 (not installed)

bin scm-4 depends on lua ~> 5.1 (5.1-1 provided by VM)
bin scm-4 depends on ffi-reloadable >= 0 (not installed)
Installing https://moonlibs.github.io/rocks/ffi-reloadable-scm-1.rockspec
Cloning into 'ffi-reloadable'...
remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 10 (delta 0), reused 1 (delta 0), pack-reused 7
Receiving objects: 100% (10/10), done.

ffi-reloadable scm-1 depends on lua >= 5.1 (5.1-1 provided by VM)
No existing manifest. Attempting to rebuild...
ffi-reloadable scm-1 is now installed in /Users/ochaton/personal/moonlibs/rocks/.rocks (license: Artistic)

env MACOSX_DEPLOYMENT_TARGET=11.0 gcc -O2 -fPIC -I/Users/ochaton/tt-apps/include/include/tarantool -c libluabin.c -o libluabin.o
env MACOSX_DEPLOYMENT_TARGET=11.0 gcc  -bundle -undefined dynamic_lookup -all_load -o libluabin-scm-4.so libluabin.o
bin scm-4 is now installed in /Users/ochaton/personal/moonlibs/rocks/.rocks (license: Artistic)
```

## What to fix

`tt rocks` should use value of the flag `--server` as "one more rocks_server" to add to manifest retrieval process

### PR DESCRIPTION
**Pull request by better0fdead**
Changed '--server' flag, so it now accepts several arguments.

Closes #686

### CODE REVIEW

File: cli/rocks/rocks.go [hunk 1]
```diff
@@ -42,7 +42,19 @@ func addLuarocksRepoOpts(cliOpts *config.CliOpts, args []string) ([]string, erro
...
 	// Check whether rocks repository is specified in tt config.
 	if cliOpts.Repo != nil && cliOpts.Repo.Rocks != "" {
-		args = append(args, "--server", cliOpts.Repo.Rocks)
+		isServerSet := false
+		for i, opt := range args {
+			if opt == "--server" {
+				isServerSet = true
+				args[i+1] = args[i+1] + " " + cliOpts.Repo.Rocks
+			} else if strings.HasPrefix(opt, "--server=") {
+				isServerSet = false
```
Comment (psergee) [inline cli/rocks/rocks.go:51]: Why is it false?

File: cli/rocks/rocks.go [hunk 2]
```diff
@@ -42,7 +42,19 @@ func addLuarocksRepoOpts(cliOpts *config.CliOpts, args []string) ([]string, erro
...
+				args[i] += " " + cliOpts.Repo.Rocks
```
Comment (psergee) [inline cli/rocks/rocks.go:52]: Shouldn't it be quoted?

File: cli/rocks/rocks_test.go [hunk 3]
```diff
@@ -72,6 +72,19 @@ func TestAddLuarocksRepoOpts(t *testing.T) {
...
+			"Rock repo is specified and --server is passed",
+			args{
+				&config.CliOpts{
+					Repo: &config.RepoOpts{
+						Rocks: "local_path",
+					},
+				},
+				[]string{"--server", "/other/repo"},
+			},
+			[]string{"--server", "/other/repo local_path"},
```
Comment (psergee) [inline cli/rocks/rocks_test.go:85]: What about `--server=`?

**Review by psergee — COMMENTED** [review]
Thank you for the patch!

File: cli/rocks/rocks.go (hunk 1 above)
Comment (better0fdead) [reply cli/rocks/rocks.go]: changed

File: cli/rocks/rocks.go (hunk 2 above)
Comment (better0fdead) [reply cli/rocks/rocks.go:52]: No, it shouldn't be quoted.
Checked it.

File: cli/rocks/rocks_test.go (hunk 3 above)
Comment (better0fdead) [reply cli/rocks/rocks_test.go:85]: Added

**Review by psergee — APPROVED** [review]
lgtm

**Pull request merged**